### PR TITLE
Changed LPFee to Percent type in Asset Conversion Pallet

### PR DESCRIPTION
**Pull request by PatricioNapoli**
Fixes https://github.com/paritytech/polkadot-sdk/issues/88

Changed Liquidity Pool fee in Asset Conversion pallet from a u32 to a Percent type.

TODO: Fix fee multipliers

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3465383

File: cumulus/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs [hunk 1]
```diff
@@ -291,6 +291,7 @@ parameter_types! {
 	pub const AllowMultiAssetPools: bool = false;
 	// should be non-zero if AllowMultiAssetPools is true, otherwise can be zero
 	pub const LiquidityWithdrawalFee: Permill = Permill::from_percent(0);
+	pub LiquidityPoolFee: Percent = Percent::from_rational_with_rounding(997u32, 1000u32, Rounding::NearestPrefDown).unwrap(); // 0.3%
```
Comment (joepetrowski) [inline cumulus/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs:294]: Won't this be zero?

Comment (PatricioNapoli) [reply cumulus/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs:294]: yeah undesired, I meant (3, 1000) which is still rounded to either 0% or 1% depending on rounding rule. I probably need more accuracy for a 0.3% using Permill (there is no inbetween Percent and Permill right?) and then the fee calculations need to take this into consideration

File: substrate/frame/asset-conversion/src/lib.rs [hunk 2]
```diff
@@ -1124,18 +1124,16 @@ pub mod pallet {
 				return Err(Error::<T>::ZeroLiquidity.into())
 			}
 
-			let amount_in_with_fee = amount_in
-				.checked_mul(&(T::HigherPrecisionBalance::from(1000u32) - (T::LPFee::get().into())))
-				.ok_or(Error::<T>::Overflow)?;
+			let fee_mult = T::HigherPrecisionBalance::from(T::LPFee::get() * 1u32);
```
Comment (PatricioNapoli) [inline substrate/frame/asset-conversion/src/lib.rs:1127]: TODO, temporal, same as below

File: cumulus/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs (hunk 1 above)
Comment (joepetrowski) [reply cumulus/parachains/runtimes/assets/asset-hub-kusama/src/lib.rs:294]: Yeah, it will have to be a `PerMill`

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3465536

File: substrate/frame/asset-conversion/src/lib.rs [hunk 3]
```diff
@@ -173,7 +173,7 @@ pub mod pallet {
 
 		/// A % the liquidity providers will take of every swap. Represents 10ths of a percent.
```
Comment (georgepisaltu) [inline substrate/frame/asset-conversion/src/lib.rs:174]:
```suggestion
		/// A % the liquidity providers will take of every swap.
```

**Comment by bkchr** [comment]
Ping @PatricioNapoli

**Pull request closed without merging**